### PR TITLE
Document repo and client's relation to MediaWiki

### PR DESCRIPTION
**Pull request by Silvan-WMDE**
This moves the reasons why Wikibase is a MediaWiki extension from the
"Architecture Decisions" to the "Solution Strategy" section. It adds a
brief explanation about the MediaWiki relation to both client's and repo's
Technical Context diagrams and summarises the whole thing on a page in
the meta "Decisions" section.

Bug: T272382

### CODE REVIEW

File: systems/WikibaseRepo/03-Context_and_Scope.md [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 ### Technical Context
 
+WikibaseRepo is being developed as a MediaWiki extension, but other than WikibaseClient, it is not a "plugin" that lives inside MediaWiki, adding functionality to it. Instead, it should be considered a system of its own, which is reflected in this diagram by not regarding MediaWiki as an external system to WikibaseRepo. See further details on this decision in the [Solution Strategy](04-Solution_Strategy.md#developing-wikibase-repo-as-a-mediawiki-extension) section.
```
Comment (tzhelyazkova) [inline systems/WikibaseRepo/03-Context_and_Scope.md:16]:
```suggestion
WikibaseRepo is being developed as a MediaWiki extension, but unlike WikibaseClient, it is not a "plugin" that lives inside MediaWiki, adding functionality to it. Instead, it should be considered a system of its own, which is reflected in this diagram by not regarding MediaWiki as an external system to WikibaseRepo. See further details on this decision in the [Solution Strategy](04-Solution_Strategy.md#developing-wikibase-repo-as-a-mediawiki-extension) section.
```

Comment (tzhelyazkova) [reply systems/WikibaseRepo/03-Context_and_Scope.md:16]: Not sure about the exact wording but "other than" sounds incorrect to me. Maybe @addshore can help.

File: systems/WikibaseRepo/09-Architecture_Decisions.md [hunk 2]
```diff
@@ -7,23 +7,3 @@ Since 2018 architecture decisions concerning Wikibase are recorded as ADRs. They
 ## Early architecture decisions
 
 This section contains retrospectively documented decisions that were made before the practice of recording them in ADRs was established. Some knowledge of the decision context likely went missing over time, so the reasoning is speculative to some degree.
-
-### Developing Wikibase Repo as a MediaWiki extension
```
Comment (tzhelyazkova) [inline systems/WikibaseRepo/09-Architecture_Decisions.md:11]: Going back to arc42's docs, it definitely makes more sense to move this whole explanation to the Solution Strategy section. :+1:

**Review by tzhelyazkova — COMMENTED** [review]
Overall looks good to me. Left a minor nitpick about wording :nerd_face:

File: systems/WikibaseRepo/03-Context_and_Scope.md (hunk 1 above)
Comment (addshore) [reply systems/WikibaseRepo/03-Context_and_Scope.md:16]: "unlike" sounds better to me!

File: docs/decisions/client-repo-mediawiki.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Client / Repo relation to MediaWiki
+
+**Decision Date:** Early 2021
```
Comment (addshore) [inline docs/decisions/client-repo-mediawiki.md:3]: I'd put January 2021 here.
The only reason I was general about the previous decisions was that it didn't seem to make sense to spend the time figuring out the exact month we decided.

File: systems/WikibaseRepo/04-Solution_Strategy.md [hunk 4]
```diff
@@ -1 +1,21 @@
...
+
+- better compatibility with the rest of the Wikimedia ecosystem
+- achieving a familiar Wikimedia user experience out of the box
+- reduction of initial boilerplate code and basic features
+- the original development team had extensive MediaWiki experience
+- ability to easily have one application that is both a client and a repo
+
+It also had negative consequences:
+
+- hacks to make MediaWiki do things it was not meant to do
```
Comment (addshore) [inline systems/WikibaseRepo/04-Solution_Strategy.md:17]: "meant to do" -> "originally designed to do"

File: systems/WikibaseRepo/04-Solution_Strategy.md [hunk 5]
```diff
@@ -1 +1,21 @@
...
+  - storing structured data in the same way Wikitext is stored
```
Comment (addshore) [inline systems/WikibaseRepo/04-Solution_Strategy.md:18]: I think we should clarify "same way" here, and say the "same place"?

This is an interesting point as this is a negative consequence that doesn't really ring true today, as MediaWiki is flexible enough to allow any content to be stored in any backend storage, we just have to hook up all of the parts (and we just haven't done it yet)

File: systems/WikibaseRepo/04-Solution_Strategy.md [hunk 6]
```diff
@@ -1 +1,21 @@
...
+  - showing different content dependent on user settings (language)
```
Comment (addshore) [inline systems/WikibaseRepo/04-Solution_Strategy.md:19]: I'm not sure if this is a negative consequence? This is desired?

**Review by addshore — APPROVED** [review]
Generally approve, but I'm keen to know what people think of my proposed tweaks

File: systems/WikibaseRepo/04-Solution_Strategy.md (hunk 6 above)
Comment (Silvan-WMDE) [reply systems/WikibaseRepo/04-Solution_Strategy.md:19]: negative or not: is it a "hack to make MediaWiki do things it was not originally designed to do"?

File: docs/decisions/client-repo-mediawiki.md (hunk 3 above)
Comment (Silvan-WMDE) [reply docs/decisions/client-repo-mediawiki.md]: done

File: systems/WikibaseRepo/04-Solution_Strategy.md (hunk 4 above)
Comment (Silvan-WMDE) [reply systems/WikibaseRepo/04-Solution_Strategy.md]: done

File: systems/WikibaseRepo/04-Solution_Strategy.md (hunk 5 above)
Comment (Silvan-WMDE) [reply systems/WikibaseRepo/04-Solution_Strategy.md]: done, and I think it's fine to leave it in here when documenting the limitations of an early solution strategy decision, even if it is not the case any longer

File: systems/WikibaseRepo/04-Solution_Strategy.md (hunk 6 above)
Comment (addshore) [reply systems/WikibaseRepo/04-Solution_Strategy.md:19]: Well, I disagree that it is a hack really.

You could say this about any framework?
No framework is designed  to magically show different output from a single block of content dependent on some user language? Whatever framework this happens in some code (not a hack) has to exist to do so.

MediaWiki allows you to render / parse a page with any number of options applied (by design).
One of these is language.
Thus different content appears depending on user settings.

This is the same mechanism that is used for rendering or not rendering the new termbox.

IMO this is not a hack, it is just a feature that we are using?
No different to having to write our own conditional parsing / rendering logic if we were using some other framework.

`request-> check user language -> render page for user using language as an option`

**Pull request merged**